src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Guid.cs

```diff
@@ -29,19 +29,25 @@ public static partial class Utf8Parser
         /// </exceptions>
         public static bool TryParse(ReadOnlySpan<byte> source, out Guid value, out int bytesConsumed, char standardFormat = default)
         {
+        FastPath:
```

Comment:
Is the generated code worse if we handle this via `case default(char)`?

I think the latter would be more readable and I would hope the JIT handles it just as nicely.

Reply:
Yes. Per the original codegen, the switch statement has some jumps before the case clauses are hit. These jumps seemingly attempt to detect if the input value is out of range early rather than fall through all the case statements one by one. Moving this to the top of the method avoids this overhead. It also avoids the overhead of needing to account for the lowercase conversion.

Reply:
> Per the original codegen, the switch statement has some jumps before the case clauses are hit. These jumps seemingly attempt to detect if the input value is out of range early rather than fall through all the case statements one by one

Ah, that is unfortunate. Maybe a comment in the code explaining that would be worthwhile? I wouldn't expect people browsing the code to look for the last PR touching the code as the first thing to understand why there is a `goto`...

> It also avoids the overhead of needing to account for the lowercase conversion

If we didn't need the goto, we could've checked against uppercase instead, which would've handled this concern